agent/app/agent_unix.go

```diff
@@ -0,0 +1,156 @@
+		// of the Agent will not be reaped leading to the ENI device
+		// disappearing until the Agent is killed.
+		return errors.New("agent is not started with an init system"), true
+	}
+
+	// Set VPC and Subnet IDs for the instance
+	if err, ok := agent.setVPCSubnet(); err != nil {
+		return err, ok
+	}
+
+	// Validate that the CNI plugins exist in the expected path and that
+	// they possess the right capabilities
+	if err := agent.queryCNIPluginsCapabilities(); err != nil {
+		// An error here is terminal as it means that the plugins
+		// do not support the ENI capability
+		return err, true
+	}
+
+	// Load the pause container's image from the 'disk'
+	if _, err := agent.pauseLoader.LoadImage(agent.cfg, agent.dockerClient); err != nil {
+		if pause.IsNoSuchFileError(err) || pause.UnsupportedPlatform(err) {
+			// If the pause container's image tarball doesn't exist or if the
+			// invocation is done for an unsupported platform, we cannot recover.
+			// Return the error as terminal for these cases
+			return fmt.Errorf("unable to load pause container image: %v", err), true
+		}
+		return fmt.Errorf("unable to load pause container image: %v", err), false
+	}
+
+	if _, err := enisetup.New(agent.ctx, state, taskEngine.StateChangeEvents()); err != nil {
+		// Inability to initialize the udev watcher could succeed if the error
+		// was due the udev socket file not being available etc
+		return fmt.Errorf("unable to set up eni watcher: %v", err), false
+	}
+
+	return nil, false
+}
+
+// setVPCSubnet sets the vpc and subnet ids for the agent by querying the
+// instance metadata service
+func (agent *ecsAgent) setVPCSubnet() (error, bool) {
+	mac, err := agent.ec2MetadataClient.PrimaryENIMAC()
+	if err != nil {
+		return fmt.Errorf("unable to get mac address of instance's primary ENI from instance metadata: %v", err), false
+	}
+
+	vpcID, err := agent.ec2MetadataClient.VPCID(mac)
+	if err != nil {
+		if isInstanceLaunchedInVPC(err) {
+			return fmt.Errorf("unable to get vpc id from instance metadata: %v", err), true
+		}
+		return instanceNotLaunchedInVPCError, false
+	}
+
+	subnetID, err := agent.ec2MetadataClient.SubnetID(mac)
+	if err != nil {
+		return fmt.Errorf("unable to get vpc id from instance metadata: %v", err), false
+	}
+	agent.vpc = vpcID
+	agent.subnet = subnetID
+	return nil, false
+}
+
+// isInstanceLaunchedInVPC returns false when the http status code is set to
+// 'not found' (404) wheb querying the vpc id from instance metadata
+func isInstanceLaunchedInVPC(err error) bool {
+	if metadataErr, ok := err.(*ec2.MetadataError); ok &&
+		metadataErr.GetStatusCode() == http.StatusNotFound {
+		return false
+	}
+
+	return true
+}
+
+// queryCNIPluginsCapabilities returns an error if there's an error querying
+// capabilities or if the required capability is absent from the capabilies
+// of the following plugins:
+// a. ecs-eni
+// b. ecs-bridge
+// c. ecs-ipam
+func (agent *ecsAgent) queryCNIPluginsCapabilities() error {
```

Comment:
Would the name`checkCNIPluginsCapabilities` or `verifyCNIPluginsCapabilities` better than `queryCNIPluginsCapabilities`?